### PR TITLE
misc sonarlint changes

### PR DESCRIPTION
**Pull request by neheb**
Mostly const changes.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: src/tiffcomposite_int.cpp [hunk 1]
```diff
@@ -488,7 +488,7 @@ TiffComponent* TiffDirectory::doAddPath(uint16_t tag, TiffPath& tiffPath, TiffCo
       return this->addNext(std::move(atc));
     return this->addChild(std::move(atc));
   }();
-  return tc->addPath(tag, tiffPath, pRoot, std::move(object));
+  return tc->addPath(tag, tiffPath, pRoot, nullptr);
```
Comment (piponazo) [inline src/tiffcomposite_int.cpp:491]: Why are you applying this change? It does not make sense at first glance 😰

**Review by piponazo — COMMENTED** [review]
Most of the changes LGTM. I only have concerns about the comment I made.

File: src/tiffcomposite_int.cpp (hunk 1 above)
Comment (neheb) [reply src/tiffcomposite_int.cpp:491]: So clang-tidy reports here (and it's correct) that there is a double move of object happening, which results in nullptr anyway. This just silences that. Actually both sonarlint and clang-tidy report it.

Comment (kevinbackhouse) [reply src/tiffcomposite_int.cpp:491]: I think clang-tidy is saying that a double move can happen _sometimes_. I don't think it's saying that `object` is _always_ null here. The `std::move` on line 477 is inside an if-statement, so it should be possible to bypass.

Comment (neheb) [reply src/tiffcomposite_int.cpp:491]: I don't think adding a null check makes sense honestly.

Comment (kevinbackhouse) [reply src/tiffcomposite_int.cpp:491]: Here's a different suggestion: #2298

Comment (neheb) [reply src/tiffcomposite_int.cpp]: Applied and removed here.

**Review by kevinbackhouse — APPROVED** [review]

**Pull request merged**